### PR TITLE
Add scheduled events data object.

### PR DESCRIPTION
**Pull request by JunSun17**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json [hunk 1]
```diff
@@ -3868,6 +3868,71 @@
       "title": "A time range.",
       "description": "For example, between 2021-05-25T13:00:00Z and 2021-05-25T14:00:00Z."
     },
+    "ScheduledEvent": {
```
Comment (yewmsft) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json:3871]: are we going to add this to stable version?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json [hunk 2]
```diff
@@ -3868,6 +3868,71 @@
...
+          "type": "string",
+          "description": "The source of a scheduled event."
+        },
+        "eventstatus": {
+          "type": "string",
+          "enum": [
+            "Scheduled",
+            "Started",
+            "Completed",
+            "Cancelled"
```
Comment (yewmsft) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json:3892]: Failed?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json [hunk 3]
```diff
@@ -3868,6 +3868,71 @@
...
+            "name": "eventstatus",
+            "modelAsString": true
+          },
+          "description": "The status of a scheduled event."
+        },
+        "scheduledtime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time of a scheduled event."
+        },
```
Comment (yewmsft) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json:3904]: Also startTime, lastUpdateTime?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json (hunk 1 above)
Comment (JunSun17) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json:3871]: I think we should do that after preview

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json (hunk 2 above)
Comment (JunSun17) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-08-01/managedClusters.json:3892]: in the message we sent, we will not know the status of maintenance event, we just tell cx what will happen.

these statuses are used in vmss events, I think it is probably better to be align with them too.

**Pull request closed without merging**